### PR TITLE
Add function TEXT #101

### LINKED ISSUE #101: Function TEXT is not implemented. TEXT is in the "Text" group
#### What actually happened

The cell contains this formula: =IF(LEN(E3)>0,AB13 TEXT(E3,"00,000"),"")
When. it evaluates, I receive the following error:

```
Traceback (most recent call last):
  File "/Users/synup/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py", line 952, in eval_func
    excel_formula.compiled_lambda())
  File "/Users/synup/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py", line 973, in <lambda>
    ### Traceback will show this line if not loaded from a text file
NameError: name 'text' is not defined
Eval: x_if(x_len(_C_("Sheet1!E3")) > 0, text(_C_("Sheet1!E3"), "00,000"), "")
Function TEXT is not implemented. TEXT is in the "Text" group
Sheet1!X3
Sheet1!Y3
Sheet1!Z3
Sheet1!AA3
Sheet1!AB3
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py in eval_func(excel_formula, cse_array_address)
    951                     ret_val = in_array_formula_context.fit_to_range(
--> 952                         excel_formula.compiled_lambda())
    953 

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py in ()
    972 
--> 973         ### Traceback will show this line if not loaded from a text file
    974 

NameError: name 'text' is not defined

During handling of the above exception, another exception occurred:

UnknownFunction                           Traceback (most recent call last)
 in 
      2 for address in cells:
      3     print(address)
----> 4     excel.evaluate(address)

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelcompiler.py in _evaluate_non_iterative(self, address)
    806                 self._gen_graph(address)
    807 
--> 808         result = self._evaluate(str(address))
    809         if isinstance(result, tuple):
    810             # trim excess dimensions

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelcompiler.py in _evaluate(self, address)
    773                 self.log.debug(
    774                     "Evaluating: {}, {}".format(cell.address, cell.python_code))
--> 775                 value = self.eval(cell)
    776                 self.log.info("Cell %s evaluated to '%s' (%s)" % (
    777                     cell.address, value, type(value).__name__))

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelcompiler.py in _eval(cell, cse_array_address)
    160                 def _eval(cell, cse_array_address=None):
    161                     return eval_ctx(
--> 162                         cell.formula, cse_array_address=cse_array_address)
    163 
    164             self._eval = _eval

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py in eval_func(excel_formula, cse_array_address)
    954             except NameError:
    955                 error_logger('error', excel_formula.python_code,
--> 956                              msg=excel_formula.msg, exc=UnknownFunction)
    957 
    958             except Exception:

~/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py in error_logger(level, python_code, msg, exc)
    904             getattr(logger, level)(error_msg)
    905             if exc is not None:
--> 906                 raise exc(error_msg)
    907             return error_msg
    908 

UnknownFunction: Traceback (most recent call last):
  File "/Users/synup/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py", line 952, in eval_func
    excel_formula.compiled_lambda())
  File "/Users/synup/anaconda3/envs/category/lib/python3.6/site-packages/pycel/excelformula.py", line 973, in 
    ### Traceback will show this line if not loaded from a text file
NameError: name 'text' is not defined
Eval: x_if(x_len(_C_("Sheet1!E3")) > 0, text(_C_("Sheet1!E3"), "00,000"), "")
Function TEXT is not implemented. TEXT is in the "Text" group
```

#### What was expected to happen
Cell should've been evaluated.

#### Problem description

  [If needed, this should explain **why** the current behavior is a problem and why the expected output is a better solution]

#### Code Sample

```python
# Your code here

filename='WORKING-tmp.xlsx'
excel = ExcelCompiler(filename=filename)
cells = [ cell.address for cell in excel.formula_cells() ]

for address in cells:
    print(address)
    excel.evaluate(address) 

```

  [If possible, include a [minimal, complete, and verifiable example](https://stackoverflow.com/help/mcve) to help us identify the issue. This also helps check that the issue is not with your own code]

#### Environment

  [Pycel Version, Python Version and OS used. Also any other environment details that you think might be relevant]

Pycel Version: python -m pip install git+https://github.com/dgorissen/pycel.git
Python Version: Python 3.6.8 :: Anaconda, Inc.
OS: Mac OS

### PR DESCRIPTION
**Pull request by Luckykarter**
- [x] closes #101
 - [x] tests 45 / 45
 - [x] passes ``tox``

### CODE REVIEW

**Comment by Luckykarter** [comment]
closed for the meantime - to make the function properly with conversion

**Comment by Luckykarter** [comment]
- [ ] closes #101
 - [ ] tests added 37 / passed 37
 - [ ] passes ``tox``

**Comment by stephenrauch** [comment]
The style changes break Flake8.  In addition they make it quite hard to review the code.  If possible can you resubmit with just the operative changes, and a successful Tox run?

**Comment by Luckykarter** [comment]
@stephenrauch, when I try to run Tox, it says 
```
tests\conftest.py:22: in <module>
    @pytest.fixture('session')
E   AttributeError: 'str' object has no attribute '__name__'
```
for all three versions of Python. Do I have something misconfigured?

**Comment by stephenrauch** [comment]
Thank you, this looks much better.  Sorry, but I have been catching up merging a bunch of work that I was neglecting for the last year +.  (Been a crazy year.)  There is now a merge conflict here.  Can you please resolve?  Also if you make an account at https://travis-ci.com/ and hook your repo fork up for testing, then it can run the TOX testing for you.

Thanks again for taking the time to submit this.

**Comment by Luckykarter** [comment]
@stephenrauch, after the last merge - tox became to work even locally. 
Though, in https://travis-ci.com/ it's passing as well 😊

Thanks!

File: tests/test_excellib.py [hunk 1]
```diff
@@ -854,6 +857,81 @@ def test_sumproduct(args, result):
...
+        ('0.244740088392962', '0%', '24%'),
+        ('0.244740088392962', '0.0%', '24.5%'),
+        ('0.244740088392962', '0.00%', '24.47%'),
+
+        # text without formatting - returned as-is
+        ('test', '', 'test'),
+        (55, '', '55'),
+
+    )
+)
```
Comment (stephenrauch) [inline tests/test_excellib.py:930]: Please consider adding some error handling and tests for same. (EG: `#DIV/0!`, `#VALUE!`, etc).

There is a decorator available to detect ERROR strings in the parameters and return same.  Its simplest form will likely work fine here:

```
@excel_helper()
def text(text_value, value_format):
    ...
```

Also a test for the error handling if you use the decorator, can be pretty easily structured like:

https://github.com/dgorissen/pycel/blob/master/tests/test_excellib.py#L656

```
def test_text_ws(fixture_xls_copy):
    compiler = ExcelCompiler(fixture_xls_copy('text.xlsx'))
    result = compiler.validate_calcs()
    assert result == {}
```

**Review by stephenrauch — CHANGES_REQUESTED** [review]
After moving these functions, please rebase and squash.  Thanks again for this contribution.

File: tests/test_excellib.py (hunk 1 above)
Comment (Luckykarter) [reply tests/test_excellib.py]: Sorry, don't have time now to prepare the *.xls for tests

**Review by stephenrauch — APPROVED** [review]

**Pull request merged**